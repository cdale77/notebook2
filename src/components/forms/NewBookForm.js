import React from "react";
import PropTypes from "prop-types";
import TextInput from "./TextInput";
import SubmitButton from "./SubmitButton";

class NewBookForm extends React.Component {
  constructor(props) {
    super(props);
    this.state = { name: "" };
  }

  handleNameChange(e) {
    this.setState({ name: e.target.value });
  }

  onSubmit(e) {
    e.preventDefault();
    this.props.onSubmit(this.state);
  }

  render() {
    return (
      <div className="form fieldset new-book-form">
        <form>
          <TextInput
            name="name"
            title="Name"
            value={this.state.name}
            onChange={this.handleNameChange.bind(this)}
          />
          <SubmitButton onSubmit={this.onSubmit.bind(this)} />
        </form>
      </div>
    );
  }
}

NewBookForm.propTypes = {
  onSubmit: PropTypes.func.isRequired
};

export default NewBookForm;
