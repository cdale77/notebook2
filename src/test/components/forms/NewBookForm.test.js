import React from "react";
import Renderer from "react-test-renderer";
import TestUtils from "react-dom/test-utils";
import NewBookForm from "../../../components/forms/NewBookForm";

describe("NewBookForm", () => {
  describe("rendering", () => {
    it("should render the form", () => {
      const component = Renderer.create(<NewBookForm onSubmit={jest.fn()} />);

      const tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

  describe("submitting the form", () => {
    it("should call the submit function", () => {
      const callBack = {
        onSubmit: () => {}
      };

      const spy = jest.spyOn(callBack, "onSubmit");
      const component = TestUtils.renderIntoDocument(
        <NewBookForm onSubmit={callBack.onSubmit} />
      );

      const button = TestUtils.findRenderedDOMComponentWithClass(
        component,
        "button"
      );

      const nameField = TestUtils.findRenderedDOMComponentWithClass(
        component,
        "input-name"
      );

      // enter values in the fields to trigger the state change callbacks
      nameField.value = "New book";
      TestUtils.Simulate.change(nameField);

      TestUtils.Simulate.click(button);

      expect(spy).toHaveBeenCalled();
      expect(callBack.onSubmit.mock.calls.length).toEqual(1);

      const form = { name: "New book" };
      expect(callBack.onSubmit.mock.calls).toEqual([[form]]);
    });
  });
});
