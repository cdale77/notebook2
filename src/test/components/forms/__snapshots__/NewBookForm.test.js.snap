// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewBookForm rendering should render the form 1`] = `
<div
  className="form fieldset new-book-form"
>
  <form>
    <div
      className="form-group"
    >
      <input
        className="input-name"
        id="name"
        name="name"
        onChange={[Function]}
        placeholder="Name"
        type="text"
        value=""
      />
    </div>
    <div
      className="form-group"
    >
      <input
        className="button"
        disabled={false}
        id="submit"
        name="submit"
        onClick={[Function]}
        type="submit"
        value="Submit"
      />
    </div>
  </form>
</div>
`;
